crates/provider/src/fillers/mod.rs

```
@@ -301,6 +315,30 @@ where
         self.inner.root()
     }
 
+    fn call<'req>(&self, tx: &'req N::TransactionRequest) -> EthCall<'req, N, Bytes> {
+        let call_filler = match self.prepare_call(tx.clone()) {
+            Ok(filled_tx) => EthCallFiller::new(filled_tx, self.weak_client()),
+            Err(e) => {
+                tracing::error!(?e, "preparing call for eth_call, using tx without filling");
+                EthCallFiller::new(tx.clone(), self.weak_client())
+            }
+        };
+        EthCall::call(call_filler, tx).block(BlockNumberOrTag::Pending.into())
+    }
+
+    fn estimate_gas<'req>(&self, tx: &'req N::TransactionRequest) -> EthCall<'req, N, U64, u64> {
+        let call_filler = match self.prepare_call(tx.clone()) {
+            Ok(filled_tx) => EthCallFiller::new(filled_tx, self.weak_client()),
+            Err(e) => {
+                tracing::error!(?e, "preparing call for eth_call, using tx without filling");
```

this silently swallows the error here.

wouldn't it be possible to move this entire logic into the `trait Caller` impl?

like a boxedfuture that does:

https://github.com/alloy-rs/alloy/blob/86f5895c55e9daeab5a19eb422b2233f78952341/crates/provider/src/provider/eth_call/caller.rs#L75-L77

but https://github.com/alloy-rs/alloy/blob/86f5895c55e9daeab5a19eb422b2233f78952341/crates/provider/src/provider/prov_call.rs#L36-L36

---

I tried moving the entier logic into the `Caller` impl by storing the `self.filler` from `FillProvider` in `EthCallFiller`.
e.g 
```rust
struct EthCallFiller<F: TxFiller<N>, N> {
filler: F
}
```
But doing this requires adding the `'static` bound to `F: TxFiller<N>` on the `FillProvider` (and up the chain into `JoinFill` as well) as `EthCall` expects `Caller` to be `'static`.

---

[a19e906](https://github.com/alloy-rs/alloy/pull/2092/commits/a19e90657dd9ec4369ad1d85c3a2f6f6a90a19d1)